### ISSUE TITLE
pool connection error error: { code: -1, message: 'Unauthenticated' } }

### ISSUE THREAD

**Issue by cron13** [cazala/coin-hive-stratum#93]
Hi, I'm constantly getting these messages:
pool connection error error:
{code: -1, message: 'Unauthenticated'}}

After restarting the proxy, everything becomes fine.
I am mine through pool.supportxmr.com 8892 and use scripts from this guide https://github.com/cazala/coin-hive-stratum/wiki/Deploy-to-now.sh-and-GitHub-Pages on frontend

**Comment by cazala** [comment]
Hi,

which version of the proxy are you using? make sure you are using the latest one `v2.6.2`.

**Comment by osfcacuk** [comment]
I have noticed that if I kill the process that is mining rather than exit sometimes the proxy thinks the miner is still connected then after a while I got the same error message. However it didn't affect miners re-joining

**Comment by cazala** [comment]
If that's the case it shouldn't impact the hashrate, but it does impact on memory (garbage collector is not purging dead miners) and bandwidth usage (if the keepalives from disconnected miners are still being sent). 

I'll look into it when I find some time. Thanks for reporting!

**Comment by tjosm** [comment]
I'm seeing this on V2.6.4

**Comment by cazala** [comment]
which pool?

**Comment by tjosm** [comment]
I'm behind XNP

**Comment by cron13** [comment]
i update proxy to v2.6.2 and problem gone

**Comment by matthewlilley** [comment]
I see this on 2.6.4 mining on Electroneum pools, not sure about Monero.

**Comment by matthewlilley** [comment]
Just checked, I see it on Monero pools also.

**Comment by patrickbau** [comment]
Can anyone help me with this error sometimes i see this error ,But i don't know how to know which version of the proxy and how to update to the version 2.6.4
Thank you

**Comment by cazala** [comment]
`npm install -g coin-hive-stratum@latest` if you use it from the CLI, or `npm install --save coin-hive-stratum@latest` if you use it from `node_modules`

**Comment by patrickbau** [comment]
Thank you 
Is that ok ?
+ coin-hive-stratum@2.6.4
added 32 packages in 3.367s

**Comment by cazala** [comment]
yes, you can check when you start the proxy it should log the version as the first line in the logs, or you can hit the `/version` endpoint while it's running to get the version as well.

**Comment by patrickbau** [comment]
Sorry i'm noob
I'm using the method Digital Ocean
I start the proxy using this command
pm2 start proxy.js --name=proxy --log-date-format="YYYY-MM-DD HH:mm"

**Comment by procloud** [comment]
something is definitely up with 2.6.6... i woke up this morning with a handful of miners connected but none where hashing.

**Comment by cazala** [comment]
have you tried `v2.6.7`?

**Comment by procloud** [comment]
i havent!

**Comment by matthewlilley** [comment]
Still present in 2.6.7, quite often I see:

"pool connection error (7dcffa94-1c68-43a0-b6d0-71488badd4d8): {"code":-1,"message":"Invalid job id"}
proxy error: undefined

**Comment by procloud** [comment]
still same issue, woke up this morning, miners connected, but not hashing with 2.6.7. no jobs coming from pool.

**Comment by matthewlilley** [comment]
There an issue mining to electroneum pools, I'd avoid it for now.
